### DISCUSSION TITLE
Quantity of quadrature points for line element

### DISCUSSION THREAD

**Discussion by andrey1887** [idaholab/moose#26098] [Q&A Modules: Solid mechanics] [answered]
Hello. `ComputeElasticityBeam::computeQpProperties()` uses quadrature point index `_qp` for assigning some properties like this `_material_stiffness[_qp](0) = _youngs_modulus[_qp];` help me please to understand where these points located.

**Comment by lindsayad** [comment, marked as answer]
The default quadrature rule is a Gaussian one. We select the number of points in the Gaussian rule in order to integrate a polynomial of order `2*p + 1` where p is the order of your finite element basis. So if you are using first order Lagrange shape functions, we would select a number of Gauss points needed to integrate a polynomial of order `2*1 + 1 = 3`. We do `2*p` to be able to integrate a mass matrix exactly and we add an additional 1 for ... well I'm still not quite firm on why we add it but @roystgnr has some explanation that involves Frobenius norms.

Anyway let's take the first Lagrange example. We want to be able to integrate a 3rd order polynomial exactly from our `2*p + 1` rule. We can do that with 2 Gauss points (see https://en.wikipedia.org/wiki/Gaussian_quadrature for details). So if using a first Lagrange basis you will see 2 quadrature points for a line element. For a quad element, the tensor product of line elements, you will see 2x2 = 4 points. A hex, 2x2x2 = 8 points.

As to *where* the points are located, the Wikipedia article is a good reference. For 2 points, on the reference element, they will be located at +/- 1/sqrt(3)

**Reply by andrey1887** [reply]
Thank you, @lindsayad. It's very helpfull information.
